### PR TITLE
[Grafana] Update grafana swagger 2022 08 01

### PR DESCRIPTION
**Pull request by lijinpei2008**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from the scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** change version of module in pull request

### CODE REVIEW

File: src/Dashboard/docs/Update-AzGrafana.md [hunk 1]
```diff
@@ -106,6 +110,22 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -GrafanaIntegrationAzureMonitorWorkspaceIntegration
+.
```
Comment (isra-fel) [inline src/Dashboard/docs/Update-AzGrafana.md:114]: Why the dot?

File: src/Dashboard/custom/New-AzGrafanaMonitorWorkspaceIntegrationObject.ps1 [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Create an in-memory object for AzureMonitorWorkspaceIntegration.
+.Description
+Create an in-memory object for AzureMonitorWorkspaceIntegration.
+
+.Outputs
+Microsoft.Azure.PowerShell.Cmdlets.Dashboard.Models.Api20220801.AzureMonitorWorkspaceIntegration
+.Link
+https://docs.microsoft.com/powershell/module/az.dashboard/new-azgrafanamonitorworkspaceintegrationobject
+#>
+function New-AzGrafanaMonitorWorkspaceIntegrationObject {
```
Comment (isra-fel) [inline src/Dashboard/custom/New-AzGrafanaMonitorWorkspaceIntegrationObject.ps1:28]: Why not use directive to generate this cmdlet?

File: src/Dashboard/README.md [hunk 3]
```diff
@@ -63,4 +63,7 @@ directive:
       verb: New
       subject: Grafana
     hide: true
+  # The cmdlet's name to long, Re-name it
+  # - model-cmdlet:
+  #     - AzureMonitorWorkspaceIntegration
```
Comment (isra-fel) [inline src/Dashboard/README.md:68]: If the commented code will not be used again, consider removing.

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/Dashboard/custom/New-AzGrafanaMonitorWorkspaceIntegrationObject.ps1 (hunk 2 above)
Comment (lijinpei2008) [reply src/Dashboard/custom/New-AzGrafanaMonitorWorkspaceIntegrationObject.ps1:28]: Monitor resource id
 "azureMonitorWorkspaceResourceId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/myResourceGroup/providers/microsoft.monitor/accounts/myAzureMonitorWorkspace"

File: src/Dashboard/README.md (hunk 3 above)
Comment (lijinpei2008) [reply src/Dashboard/README.md:68]: After a command is renamed, the command name is different from the original one. Therefore, retain comments for subsequent maintenance

File: src/Dashboard/docs/Update-AzGrafana.md [hunk 4]
```diff
@@ -154,6 +158,22 @@ Accept pipeline input: True (ByValue)
 Accept wildcard characters: False
 ```
 
+### -MonitorWorkspaceIntegration
+.
```
Comment (isra-fel) [inline src/Dashboard/docs/Update-AzGrafana.md:162]: Here is an extra dot.

Comment (lijinpei2008) [reply src/Dashboard/docs/Update-AzGrafana.md]: Done.

**Review by isra-fel — COMMENTED** [review]
Hi @lijinpei2008 I had a discussion with Xigaogang, a more robust solution is to update swagger using directives. Let me know if you have any concern.

**Comment by lijinpei2008** [comment]
> Hi @lijinpei2008 I had a discussion with Xigaogang, a more robust solution is to update swagger using directives. Let me know if you have any concern.

Do I have to wait for the AutoRest script to upgrade?
The way I handle it now is to make changes in a custom folder

**Review by isra-fel — APPROVED** [review]

**Pull request merged**